libs/database/src/user.rs

```diff
@@ -39,20 +39,26 @@ pub async fn update_user(
   if let Some(n) = name {
     args_num += 1;
     set_clauses.push(format!("name = ${}", args_num));
-    args.add(n);
+    args
+      .add(n)
+      .map_err(|err| AppError::SqlxError(err.to_string()))?;
   }
 
   if let Some(e) = email {
     args_num += 1;
     set_clauses.push(format!("email = ${}", args_num));
-    args.add(e);
+    args
```

Comment:
The problem is, the .add method returns a generic error, rather than sqlx specific error type. So I am not sure how can we auto convert it to AppError.

Reply:
https://github.com/launchbadge/sqlx/blob/572e2a4ed533a0ef242e946443b14bdb9cc91bc4/sqlx-core/src/arguments.rs#L20

Reply:
I can map the error to a more specific type of error, instead of SqlxError. From what I understand this only happens if there is any encoding issue, like a malformed json string.

Reply:
@appflowy How about something like:
```
    args.add(n).map_err(|err| AppError::SqlxArgEncodingError {
      desc: format!("unable to encode user name"),
      err,
    })?;
```

Reply:
LGTM